api/client/build.go

```diff
@@ -265,6 +268,14 @@ func (cli *DockerCli) CmdBuild(args ...string) error {
 	headers.Add("X-Registry-Config", base64.URLEncoding.EncodeToString(buf))
 	headers.Set("Content-Type", "application/tar")
 
+	// collect all the build-time environment variables for the container
+	buildArgs := runconfig.ConvertKVStringsToMap(flBuildArg.GetAll())
+	buf, err = json.Marshal(buildArgs)
+	if err != nil {
+		return err
+	}
+	headers.Add("X-Docker-BuildArgs", base64.URLEncoding.EncodeToString(buf))
```

Comment:
Thanks @cpuguy83 ! (I started looking around for something to support https://github.com/docker/docker/pull/15182#discussion_r36207735 in Go, but not enough into Go yet, and couldn't find a proper example)